### PR TITLE
Uast: `KotlinUNestedAnnotation` for processing nested annotations (IDEA-185890)

### PR DESCRIPTION
**Pull request by xiexed**

### CODE REVIEW

File: compiler/light-classes/src/org/jetbrains/kotlin/asJava/elements/lightAnnotations.kt [hunk 1]
```diff
@@ -255,6 +260,25 @@ class KtLightAnnotationForSourceEntry(
 
     override fun findDeclaredAttributeValue(name: String?) = clsDelegate.findDeclaredAttributeValue(name)?.let { wrapAnnotationValue(it) }
 
+    override fun getParameterList(): PsiAnnotationParameterList = KtLightAnnotationParameterList(super.getParameterList())
+
+    inner class KtLightAnnotationParameterList(private val list: PsiAnnotationParameterList) : KtLightElementBase(this),
+        PsiAnnotationParameterList {
```
Comment (yanex) [inline compiler/light-classes/src/org/jetbrains/kotlin/asJava/elements/lightAnnotations.kt]: Formatting: `{` should be on the next line.

File: compiler/light-classes/src/org/jetbrains/kotlin/asJava/elements/lightAnnotations.kt [hunk 2]
```diff
@@ -255,6 +260,25 @@ class KtLightAnnotationForSourceEntry(
...
+        override val kotlinOrigin get() = null
+        override fun getAttributes(): Array<PsiNameValuePair> = list.attributes.map { KtLightPsiNameValuePair(it) }.toTypedArray()
+
+        inner class KtLightPsiNameValuePair(private val psiNameValuePair: PsiNameValuePair) : KtLightElementBase(this),
+            PsiNameValuePair {
```
Comment (yanex) [inline compiler/light-classes/src/org/jetbrains/kotlin/asJava/elements/lightAnnotations.kt]: `{`: the same.

File: plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/declarations/KotlinUAnnotation.kt [hunk 3]
```diff
@@ -93,3 +101,32 @@ class KotlinUAnnotation(
...
+    override fun annotationUseSiteTarget() = annotationEntry.useSiteTarget?.getAnnotationUseSiteTarget()
+
+}
+
+class KotlinUNestedAnnotation(
+    original: KtCallExpression,
+    givenParent: UElement?,
+    private val classDescriptor: ClassDescriptor?
+) : KotlinUAnnotationBase(original, givenParent) {
+    override val javaPsi: PsiAnnotation? = null
```
Comment (yanex) [inline plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/declarations/KotlinUAnnotation.kt]: Why `null`?

Comment (yanex) [reply plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/declarations/KotlinUAnnotation.kt]: `annotationEntry.toLightAnnotation()`?

File: compiler/light-classes/src/org/jetbrains/kotlin/asJava/elements/lightAnnotations.kt (hunk 1 above)
Comment (xiexed) [reply compiler/light-classes/src/org/jetbrains/kotlin/asJava/elements/lightAnnotations.kt]: reformat reverts it back to current state

**Pull request merged**